src/SFTPFile.cpp

```diff
@@ -11,6 +11,7 @@
 #include <kodi/addon-instance/VFS.h>
```

Comment:
About:
> Azure pipeline are complaining about incorrect overloading for the function RemoveDirectory and CreateDirectory, any idea why is that occuring? I do not have such failure while building on Linux.

Is a typical Windows pain in the ass problem, as them brings macros with names used as function in API. For here the used `CreateDirectory` and `DeleteFile`. Here the addon header to prevent the problems https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-dev-kit/include/kodi/c-api/filesystem.h#L22-L32.

One solution can be to move `#include <kodi/addon-instance/VFS.h>` below `<fcntl.h>` (seems this fcntl.h brings back the bad Windows macros).

Another can be to add this again below `<fcntl.h>`:
```cpp
#ifdef CreateDirectory
#undef CreateDirectory
#endif // CreateDirectory
#ifdef DeleteFile
#undef DeleteFile
#endif // DeleteFile
```
```

Reply:
I made that change (include kodi below fcntl.h) but the azure pipeline is still failling

Reply:
Builds are succesfull now with the exception of iOS which fails on linking libssh.a, not sure why actually, any idea?

Reply:
About iOS and also tvOS it is extremely strange by this addon, still not found why, but a lot of times it fails with:
```
error: /Applications/Xcode11.3.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: can't open file: libssh.a
```
Is since a very long time and also tried to becomes fixed and every attempt not help. The only solution is to trigger the builds after a release so often until all was OK (also tried with this https://github.com/xbmc/vfs.sftp/pull/55).

Reply:
Indeed, I noticed the same looking at the previous commit on the Nexus branch.

Let me know if anything is required before merging.